File: src/main/java/nz/co/aetheric/sameness/equatators/SimpleEquatator.java
package nz.co.aetheric.sameness.equatators;

import javax.validation.constraints.NotNull;

/**
 * TODO: What is the purpose of this class?
 * <p>Author: <a href="http://gplus.to/tzrlk">Peter Cummuskey</a></p>
 */
public class SimpleEquatator<T> extends NullsafeEquatator<T> {

	public static <T> SimpleEquatator<T> simpleEquatator(Class<T> type) {
		return new SimpleEquatator<>(type);
	}

	@SuppressWarnings("WeakerAccess")
	public SimpleEquatator(@NotNull Class<T> type) {
		super(type);
	}

	@Override
	public <L extends T, R extends T> boolean isEquatable(L left, R right) {
		return super.bothNull(left, right) || super.neitherNull(left, right)
				&& left.equals(right);
	}

}

File: src/main/kotlin/nz/co/aetheric/sameness/equatators/SimpleEquatator.kt
package nz.co.aetheric.sameness.equatators

/**
 * TODO: What is the purpose of this class?
 *
 * Author: [Peter Cummuskey](http://gplus.to/tzrlk)
 */
class SimpleEquatator<T>(type: Class<T>) : NullsafeEquatator<T>(type) {

	companion object {

		fun <T> simpleEquatator(type: Class<T>)
				= SimpleEquatator(type)

	}

	override fun <L : T, R : T> isSafelyEquatable(left: L, right: R)
			= left == right

}
